Lib/importlib/readers.py

```diff
@@ -7,11 +7,19 @@ class FileReader(abc.TraversableResources):
     def __init__(self, loader):
         self.path = pathlib.Path(loader.path).parent
 
+    def resource_path(self, resource):
+        """
+        Return the file system path to prevent
+        `resources.path()` from creating a temporary
+        copy.
+        """
+        return str(self.path.joinpath(resource))
+
     def files(self):
         return self.path
 
 
-class ZipReader(FileReader):
+class ZipReader(abc.TraversableResources):
```

Comment:
Is this a safe (as in backward-compatible) change?

Reply:
Yes. The main reason this was here was to supply FileReader.files, but now it has its own copy.

Reply:
Yes. The reason this was present was to re-use FileReader.files, but now it's native.